sycl/doc/extensions/proposed/sycl_ext_oneapi_queue_priority.asciidoc

```diff
@@ -0,0 +1,102 @@
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2022-2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 5 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+== Specification
+
+This extension defines additional queue properties that permit a backend,
+when supported, to create command queues with different levels of priority.
+Priorities are hints from the programmer that certain command submissions 
+should execute before others of lesser priority if the backend must choose
+between different commands ready to execute.
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_QUEUE_PRIORITY` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+=== API Additions
+
+
+This extension defines the following new properties that may be passed to the 
+constructor of the `queue` class:
+
+[%header,cols="1,5"]
+|===
+|Property
+|Description
+
+| `sycl::ext::oneapi::property::queue::low_priority`
+| This property signals that commands submitted to this queue should be treated with
+less priority than all others.
+
+| `sycl::ext::oneapi::property::queue::normal_priority`
+| This property signals that commands submitted to this queue should be treated with
+the normal level of priority. This is the default behavior, so passing this property
+is not required.
+
+| `sycl::ext::oneapi::property::queue::high_priority`
+| This property signals that commands submitted to this queue should be treated with 
+more prioirty than all others.
+|===
+
```

Comment:
Two thoughts came to mind after reading this:

* This extension uses the SYCL 2020 properties rather than the new compile-time properties.  Is that what we want?  One reason *not* to use the compile-time properties is because they are experimental.  If we want this extension to be "supported", then we should not use the compile-time properties.

* Do we think people will be happy with the low / medium / high levels?  Will people ask for numerical levels?  I think this might be easier with compile-time properties, so maybe we would consider this if / when we move to that.